app/src/business_logic_layer/services/coordinate.service.ts

```diff
@@ -1,12 +1,15 @@
 import { PrismaClient, Coordinate, Obstacle } from '@prisma/client';
 import { CoordinateRepository, SessionRepository } from '../../data_access_layer/repositories';
+import { BadRequestError, InternalServerError, NotFoundError } from '../../common/errors';
 export class CoordinateService {
     constructor(private coordinateRepository: CoordinateRepository, private sessionRepository: SessionRepository) {}
 
     public async createCoordinate(sessionId: string, x: number, y: number): Promise<Coordinate> {
         try {
             const timestamp = new Date();
             const coordinate: Coordinate = await this.coordinateRepository.createPosition(sessionId, x, y, timestamp);
+            if (!coordinate) throw new InternalServerError("Coordinate not created");
```

Comment:
ehh seems not correct, is it really a server error? In what case would coordinateRepository.createPosition() return null? You can leave it at that but fix it later.